server/lists/listController.js

```diff
@@ -113,38 +113,41 @@ module.exports = {
     });
   },
 
-  deleteItemFromList: function(req, res) {
+  updateItem: function(req, res) {
     var username = interimUsername;
-    var itemName = req.body.name.toLowerCase();
+    var newName = req.body.name.toLowerCase();
+    var index = req.body.index;
+
+    var setModifier = { $set: {} };
+    setModifier.$set['list.' + index] = req.smartShoppingData._id;
+    User.findOneAndUpdate({username: username}, setModifier, {upsert: true}, function(err, user) {
+      if (err) {
+        console.log(err);
+        res.status(500).send({error: 'Server Error'});
+      } 
+      res.send(user.list);
+    }); 
+  },
 
+  deleteItemFromList: function(req, res) {
+    var username = interimUsername;
+    var index = Number(req.body.index);
     
-    var findUser = Q.nbind(User.findOne, User);
-    var findItem = Q.nbind(Item.findOne, Item);
+    var setModifier = { $set: {} };
+    setModifier.$set['list.' + index] = null;
+    User.update({username: username}, setModifier, {upsert: true}, function(err) {
+      if (err) {
+        console.log(err);
+        res.status(500).send({error: 'Server Error'});
+      } 
+    });
 
-    findItem({name: itemName})
-    .then(function(match) {
-      findUser({username: username})
-      .then(function(user) {
-        User.findByIdAndUpdate(
-          user._id,
-          {$pull: {'list': match._id}},
-          {safe: true, upsert: true},
-          function(err, model) {
-            if(err) console.error(err);
-          }
-        );
-      res.send(user.list);   
-      })
-      .catch(function(err) {
-        console.error(err);
-        res.status(500).send({ error: 'Server Error' });
-      })
-    })
-    .done(function(err) {
+    User.findOneAndUpdate({username: username}, {$pull: {'list': null}}, {upsert: true}, function(err, user) {
```

Comment:
Since the `User.update` call on line 138 is asynchronous, is it guaranteed that the item will be set to null when this filter is run? Maybe converting `User.update` to a promise with `Q.nbind` and then putting the `User.findOneAndUpdate` pseudo-filter call in a `.then` callback would guarantee that?